src/gui/main.cpp

```diff
@@ -48,43 +48,16 @@ int main(int argc, char **argv)
     Q_INIT_RESOURCE(client);
 
     QCoreApplication::setAttribute(Qt::AA_UseHighDpiPixmaps, true);
-#ifdef Q_OS_WIN
-    QCoreApplication::setAttribute(Qt::AA_EnableHighDpiScaling, true);
-#endif // !Q_OS_WIN
 
-#ifdef Q_OS_MAC
-    Mac::CocoaInitializer cocoaInit; // RIIA
-#endif
+    // Create a `Platform` instance so it can set-up/tear-down stuff for us, and do any
+    // initialisation that needs to be done before creating a QApplication
+    QScopedPointer<Platform> platform(Platform::create());
```

Comment:
A scoped pointer and a singleton?
If we really need the singleton should we rather use a qAddPostRoutine ?